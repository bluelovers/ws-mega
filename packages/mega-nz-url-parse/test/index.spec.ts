import parseMegaLink from '../index';

test(`parseMegaLink 01`, () =>
{
	let href = `https://mega.nz/folder/dw4DzZhJ#RNFlsWOf-QTOZJvsMmqLlA/file/5hBUQarQ`

	let actual = parseMegaLink(href);
	let expected;

	//expect(actual).toStrictEqual(expected);
	//expect(actual).toBeInstanceOf(Date);
	expect(actual).toMatchSnapshot();

});

test(`parseMegaLink 02`, () =>
{
	let href = `https://mega.nz/folder/dw4DzZhJ#RNFlsWOf-QTOZJvsMmqLlA`

	let actual = parseMegaLink(href);
	let expected;

	//expect(actual).toStrictEqual(expected);
	//expect(actual).toBeInstanceOf(Date);
	expect(actual).toMatchSnapshot();

});

test(`parseMegaLink 03`, () =>
{
	let href = `https://mega.nz/file/dw4DzZhJ#RNFlsWOf-QTOZJvsMmqLlA`

	let actual = parseMegaLink(href);
	let expected;

	//expect(actual).toStrictEqual(expected);
	//expect(actual).toBeInstanceOf(Date);
	expect(actual).toMatchSnapshot();

});
