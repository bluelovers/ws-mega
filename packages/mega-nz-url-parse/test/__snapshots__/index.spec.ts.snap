// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseMegaLink 01 1`] = `
Object {
  "root": Object {
    "directory": true,
    "downloadID": "dw4DzZhJ",
    "key": "RNFlsWOf-QTOZJvsMmqLlA",
  },
  "sub": Object {
    "directory": false,
    "downloadID": "5hBUQarQ",
  },
  "url": "https://mega.nz/folder/dw4DzZhJ#RNFlsWOf-QTOZJvsMmqLlA/file/5hBUQarQ",
}
`;

exports[`parseMegaLink 02 1`] = `
Object {
  "root": Object {
    "directory": true,
    "downloadID": "dw4DzZhJ",
    "key": "RNFlsWOf-QTOZJvsMmqLlA",
  },
  "url": "https://mega.nz/folder/dw4DzZhJ#RNFlsWOf-QTOZJvsMmqLlA",
}
`;

exports[`parseMegaLink 03 1`] = `
Object {
  "root": Object {
    "directory": false,
    "downloadID": "dw4DzZhJ",
    "key": "RNFlsWOf-QTOZJvsMmqLlA",
  },
  "url": "https://mega.nz/file/dw4DzZhJ#RNFlsWOf-QTOZJvsMmqLlA",
}
`;
